src/main/java/org/opensearch/knn/index/util/Faiss.java

```diff
@@ -226,7 +228,28 @@ class Faiss extends NativeLibrary {
                     return ((4L * centroids * dimension) / BYTES_PER_KILOBYTES) + 1;
                 })
                 .build()
-        ).addSpaces(SpaceType.L2, SpaceType.INNER_PRODUCT).build()
+        ).addSpaces(SpaceType.L2, SpaceType.INNER_PRODUCT).build(),
+        METHOD_NSG,
+        KNNMethod.Builder.builder(
+            MethodComponent.Builder.builder(METHOD_NSG)
+                .addParameter(
+                    METHOD_PARAMETER_M,
+                    new Parameter.IntegerParameter(METHOD_PARAMETER_M, KNNSettings.INDEX_KNN_DEFAULT_ALGO_PARAM_M, v -> v > 0)
+                )
+                .addParameter(
+                    METHOD_ENCODER_PARAMETER,
+                    new Parameter.MethodComponentContextParameter(METHOD_ENCODER_PARAMETER, ENCODER_DEFAULT, encoderComponents)
+                )
+                .setMapGenerator(
+                    ((methodComponent, methodComponentContext) -> MethodAsMapBuilder.builder(
+                        FAISS_NSG_DESCRIPTION,
+                        methodComponent,
+                        methodComponentContext
+                    ).addParameter(METHOD_PARAMETER_M, "", "").addParameter(METHOD_ENCODER_PARAMETER, ",", "").build())
+                )
+                .build()
+        ).addSpaces(SpaceType.L2, SpaceType.COSINESIMIL).build()
```

Comment:
I believe this should be inner product?

Reply:
++ FIXED at https://github.com/opensearch-project/k-NN/pull/945/commits/3c8203be7edf96757f76a1cb18a48823980c08d9